enhancements/microshift/microshift-router-configuration-performance.md

```diff
@@ -0,0 +1,199 @@
+---
+title: microshift-router-configuration-performance
+authors:
+  - "@eslutsky"
+reviewers:
+  - "@pacevedom"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+  - "@Miciah"
+approvers:
+  - "@jerpeter1"
+api-approvers:
+  - None
+creation-date: 2024-09-23
+last-updated: 2024-10-07
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-4091
+---
+
+# MicroShift router Operations & performance configuration options
+
+## Summary
+MicroShift's default router is created as part of the platform, but does not
+allow configuring any of its specific parameters. For example, you cannot
+specify the policy for HTTP traffic compression or enable HTTP2 protocol.
+
+In order to allow these operations and many more, a set of configuration options
+is proposed.
+
+## Motivation
+
+Microshift Customers need a way to override the default Ingress Controller  configuration parameters similar as OpenShift does.
+
+
+### User Stories
+
+#### User Story 1
+
+My application starts processing requests from clients, but the connection is
+getting closed before it can respond.
+
+I set `ingress.tuningOptions.serverTimeout` in the configuration file to a
+higher value to accommodate the slow response from the server.
+
+#### User Story 2
+
+The router has many connections open because an application running on my
+cluster doesn't close connections properly.
+
+I set `ingress.tuningOptions.serverTimeout` and
+`spec.tuningOptions.serverFinTimeout` in the ingresscontroller API to a lower
+value, forcing those connections to close sooner if my application stops
+responding to them.
+
+### Goals
+Allow users to configure the additional HAProxy/Router performance customization parameters, see Proposal table for details.
+
+
+### Non-Goals
+N/A
+
+## Proposal
+
+Microshift don't use ingress [operator](https://github.com/openshift/cluster-ingress-operator) , all the customization performed through  configuration file.
+the configuration will propagate to the router deployment [manifest](https://github.com/pacevedom/microshift/blob/8f76e21b9a3f0044c83eae4e2c177871c8235103/assets/components/openshift-router/deployment.yaml) Environment variables.
+
+see the table below for the proposed configuration changes:
```

Comment:
Should we reference OCP docs here? You would get extra info such as possible values, format, etc.
https://docs.openshift.com/container-platform/4.17/networking/ingress-operator.html

Reply:
I think the table is good way to show which configuration we are adding in the enhancement.
i'll also add reference to the full OCP config.